libvast/src/transform.cpp

```diff
@@ -21,7 +22,7 @@ caf::expected<table_slice> transform_step::apply(table_slice&& slice) const {
   const auto* generic_step = dynamic_cast<const generic_transform_step*>(this);
   const auto* arrow_step = dynamic_cast<const arrow_transform_step*>(this);
   VAST_ASSERT(arrow_step || generic_step);
-  if (arrow_step)
+  if (arrow_step && slice.encoding() == table_slice_encoding::arrow)
```

Comment:
Should this be documented as a separate bug fix? Not sure how this worked before.

Reply:
Is a separate commit adequate for this purpose?

Reply:
Create as many commits as you like :)